### PR TITLE
Add and rearange initial example

### PR DESCRIPTION
**Pull request by paulf81**
This pull requests combines former branches which added new initial examples, as well as interactive streamlit demonstrations.

Finally pull requests adds a new notebook which maybe could provide a first overview in critical syntax and how some things are importantly changed from v2 and v3

This pull request also seeks to address issue #318 and issue #315 by providing clearer demonstration of recommended use via the above notebook.

Not yet, still some discussion

**Feature or improvement description**
Adds four additional examples:
examples/__getting_started.ipynb
examples/00_getting_started.py
examples/12_streamlit_demo.py
examples/13_streamlit_demo_large_farm.py

However the main idea is to agree on the syntax rules laid out in the notebook, then all following examples will be updated

**Related issue, if one exists**
 issue #318
issue #315

### CODE REVIEW

**Comment by rafmudaf** [comment]
It's great that there's a notebook explaining things slowly and showing the value of variables. Excellent idea @paulf81.

On that notebook, my only feedback is that the tools API will certainly change over the next few days so we'll have to be sure to update the example and explanations.

**Comment by rafmudaf** [comment]
> should I go through and update the other examples to follow this syntax (copying the FI for plotting)?

I'm in the process of redeveloping the `floris.tools` package now so these API's and patterns will change very soon. I vote lets wait until that's nailed down and then update the examples.

**Comment by paulf81** [comment]
hi @bayc  and @rafmudaf  I went through all the examples after merging in the latest v3, and updated the visualization methods to the new standards/function name and confirm each example works.  I think this doing a better and better job of introducing the key concepts in an incremental way, but will be curious what you (and also @Bartdoekemeijer ) think

File: examples/03_making_adjustments.py [hunk 1]
```diff
@@ -35,27 +35,27 @@
 # Initialize FLORIS with the given input file via FlorisInterface
 fi = FlorisInterface("inputs/gch.yaml")
 
-# Configure this simulation for visualization using the full flow field grid
-solver_settings = {
-    "type": "flow_field_grid",
-    "flow_field_grid_points": [200,100,7]
-}
-fi.reinitialize(solver_settings=solver_settings)
+# # Configure this simulation for visualization using the full flow field grid
```
Comment (bayc) [inline examples/03_making_adjustments.py:38]: Should this just be removed? Is it not needed anymore?

File: examples/12_streamlit_demo.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+st.header("Visualizations")        
+st.write(fig_viz)
+st.header("Power Comparison")
+st.write(fig_turb_pow)
+
+# print(farm_power_results)
+
+# # Show the farm results
+# df_farm = pd.DataFrame.from_records(farm_power_results, columns = ['model','yaw','power']).set_index(['yaw','model']).unstack()
```
Comment (bayc) [inline examples/12_streamlit_demo.py:171]: Should this code be removed?

File: examples/12_streamlit_demo.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+        horizontal_plane_gch = fi.calculate_horizontal_plane(x_resolution=100, y_resolution=100, yaw_angles=yaw_angles_yaw)
+        visualize_cut_plane(horizontal_plane_gch, ax=ax, title='%s - yawed' % fm)
+
+st.header("Visualizations")        
+st.write(fig_viz)
+st.header("Power Comparison")
+st.write(fig_turb_pow)
+
+# print(farm_power_results)
```
Comment (bayc) [inline examples/12_streamlit_demo.py:168]: Remove?

File: examples/12_streamlit_demo.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
+# Copyright 2021 NREL
```
Comment (bayc) [inline examples/12_streamlit_demo.py:1]: Should this be renamed to example 11?

File: examples/03_making_adjustments.py (hunk 1 above)
Comment (paulf81) [reply examples/03_making_adjustments.py:38]: done!

File: examples/12_streamlit_demo.py (hunk 2 above)
Comment (paulf81) [reply examples/12_streamlit_demo.py:171]: Done!

File: examples/12_streamlit_demo.py (hunk 3 above)
Comment (paulf81) [reply examples/12_streamlit_demo.py:168]: Done!

File: examples/12_streamlit_demo.py (hunk 4 above)
Comment (paulf81) [reply examples/12_streamlit_demo.py:1]: or we can save 11 for a mulit turbine example?

**Pull request merged**